### DISCUSSION TITLE
Modbus zu WiFi/MQTT

### DISCUSSION THREAD

**Discussion by ponchofiesta** [Tom-Bom-badil/samson_trovis_557x#42] [General (de/en)]
Hallo zusammen,
ich möchte meine Trovis-Anlage 5573 in Home Assistant einbinden.
Soweit ich es verstanden hab, müsste es damit doch gehen:
https://www.amazon.de/gp/product/B0CG173Y78

Damit würde ich die Anlage per MQTT über WiFi in HA einbinden.

Hat das schon jemand gemacht?
Hat jemand einen Tipp, wie ich das Gerät an das Trovis anschließe? Brauche ich da vielleicht noch einen Adapter von den einzelnen Pins der Trovis zur RS232-Buchse des Modbus-Adapters?

**Comment by tugsi** [comment]
Hallo @ponchofiesta ,
wenn Du etwas basteln magst und kannst, dann kannst etwas Geld sparen, da der Waveshare schon recht teuer ist.
Du kannst ein ESP32 nehmen und diesen mit ESPHome bespielen und dann per MQTT an den Broker schicken. 

Wenn den 5573-0001 hast, dann brauchst allerdings noch ein Levelshifter dazu, Infos findest hier [https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/13](url)

Ansonsten habe ich selber mit einem schönen kleinen ESP32-C6 gemacht, näheres dazu findest Du hier: [https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/6#discussioncomment-9831221](url)
und weiter unten im Thread. 
Den C6 hab ich günstig bei Botland bekommen.

Da hast erstmal was zu lesen  :-)

**Comment by Tom-Bom-badil** [comment]
@tugsi @tmycgn
Mal so aus Interesse, da ich in meinen diesjährigen Winterprojekten auch mal wieder meine ESP-Sammlung erneuern und erweitern will (hab da jahrelang nix dran gemacht - die Dinger laufen einfach und messen fleißig Temp/Feuchte/Druck in Haus und Schuppen und übergeben ihre Daten alle 5 Minuten an eine Rest API für SmartHomeNG + HA):

ESP's waren ja schon immer dauerklamm in Sachen Speicher und Verarbeitungsgeschwindigkeit. Habt Ihr schon Erfahrungen hinsichtlich:

- der maximalen Anzahl an Entitäten, die man aus Speichergründen einem ESP mit Modbus-Abindung antun kann,
- der maximalen Anzahl an Entitäten, die man aus Performance-Gründen (CPU, Wifi, Protokoll-Overhead) einem ESP mit Modbus-Abindung antun kann?

Hintergrund: Die Zahl der Trovis-Entitäten im HA Projekt liegt schon deutlich über 400 - manche werden alle 60s abgefragt, manche alle 15 Minuten. Ich hab so die Befürchtung, da irgendwo an ESPHome-Grenzen zu stoßen ...

/tom

**Reply by tugsi** [reply]
@Tom-Bom-badil 
Da muss ich leider passen, ich benutze ESP momentan auch nur in kleinen Sachen, ähnlich wie Du.
Mit ESPHome kam ich jetzt erst in Berührung, hatte zwar schon von gehört, aber nie wirklich beachtet, zumal ich ioBroker benutze und nicht HomeAssistant, wo ESPHome eher zuhause ist. 
Bisher war es Tasmota bei mir, allerdings muss ich sagen, ESPHome hat definitiv seinen Reiz, hab mir auf meinen Proxmox ein LXC esphome gemacht um den ESP C6 hier anzubinden, klappt super, die Daten schicke ich mir aber per MQTT zum Broker, wo dann mein ioBroker draufschaut. 

Allerdings sind die ESP schon viel besser geworden, also nicht mehr vergleichbar mit "nur" einem ESP8266, die ESP32-Reihe kann schon viel [https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf](url)

Eventuell muss man es einfach testen, was er schafft.
Ich weiß nur nicht, wie man es beim ESPHome einfach und ohne Fleißarbeit, alle Modbus-Adressen einbaut.

**Reply by tmycgn** [reply]
Dito, ebenso. Ich denke aber das der ESP32 schon einiges verarbeitet bekommt und hätte da bei deinen erwähnten Mengen keine Bedenken. Hab gestern noch meine Wasseruhr für rund 10 EUR smart gemacht, klappt einfach super mit Esphome und den NanoC6. Also lohnt sich da dranzubleiben. ;)

**Reply by tugsi** [reply]
Mit dem C6? Welche Cam hast Du denn benutzt dafür?

**Reply by tmycgn** [reply]
Nein, mit einem LJ18A3-8-Z/BX 5V Näherungssensor direkt an den Grove Port. Der zählt einfach den Impuls von der Mini-Metalplatte im Zählen. (1 Umdrehung = 1 l) Siehe auch https://github.com/geertmeersman/energie-meter

**Reply by tugsi** [reply]
Ah interessant, ich kannte bisher nur ein Projekt mit Cam, welches dann ein Tool auswertet.
Hatte mal gerade nach den Näherungssensor geschaut, den gibt's auch als NPN etwas kleiner als LJ12, theoretisch müsste der auch gehen, muss ja nur das Rad erfassen. 
Den LJ18 finde ich sehr oft als 6-36V-Variante, habe schon den Verdacht, dass es theoretisch immer derselbe ist, auch wenn da 5V extra steht, kannst Du da was zu sagen.
Wie hast Du den fixiert, mit einem Bügel oder etwas 3D-gedrucktem? 
Achso, wenn ich gerade dabei bin mit Fragen stellen, wie löst Du elegant das Spannungsproblem, hast Du irgendwas kleines auf USB-C oder nimmst normale Netzteile, wie für´s Handy?

**Reply by tmycgn** [reply]
Hab den oben auf Empfehlung genommen, gibts bei Amaz* und Ali. Muss aber exakt die 5V Version sein. Fixiert noch mit Tape, wollte mir evtl. noch was drucken. Gibt da auch schon Vorlagen auf den bekannten Portalen. Netzteil nehme ich für die IoT Dinge lieber was "Gutes" "Sicheres" und hab gute Erfahrung hiermit gemacht: "Brennenstuhl USB Steckdosenadapter (USB Ladegerät zum Einstecken in ein Steckdosennest, 2,4A, kompatibel mit IP44 Steckdosen und Kabeltrommeln von Brennenstuhl, flaches Design)". Sehr flach und wird nicht warm. Im Zählerschrank hab ich mir dann noch einen 90 Grad USB Winkel gegönnt, dann hast du es noch flacher. :)
